File: NetBeansProjects/Actividad12/src/actividad12/Actividad12.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package actividad12;

import javax.swing.JOptionPane;

/**
 *
 * @author andga
 */
public class Actividad12
{

	/**
	 * @param args the command line arguments
	 */
	public static void main(String[] args)
	{
		int numInput, digitos, resultPowMax, ultDig1_resultPot, ultDig2_resultPot, ultimos2Dig, cantDigResultPot;
		String numInputText, resultPotText, sResultPot, ultimos2DigitosResPot = "";
		boolean igualDigito, automorfico;
		long resultPot;

		numInput = Integer.parseInt(JOptionPane.showInputDialog("Ingrese un numero:"));
		//System.out.println("User input: " + numInput);

		resultPowMax = (int) Math.pow(numInput, numInput);
		//System.out.println("resultado potencia maxima: " + resultPowMax);
		numInputText = String.valueOf(numInput);
		//System.out.println("User Input int Cambia tipo de datos a String: " + numInputText);
		resultPotText = String.valueOf(resultPowMax);
		//System.out.println("Resultado potencia int Cambia tipo de datos a String: " + resultPotText);
		digitos = numInputText.length();
		//System.out.println("Cantidad digitos \"length\" de Input User: " + digitos);
		
		//CalcularPotencia()
		System.out.println("Base | Exponente | Resultado ");
		for (int i = 1; i <= numInput; i++) {
			resultPot = (long) Math.pow(numInput, i);
			sResultPot = String.valueOf(resultPot);
			System.out.print(numInput + "    | " + i + "\t | " + resultPot + " |");

			if (digitos == 1) {
				//Cantidad digitos de Resultado Potencia
				cantDigResultPot = sResultPot.length();

				//Extraer ultima digito de Resultado Potencia "String" a variable ultDig_resultPot
				ultDig1_resultPot = Character.getNumericValue(sResultPot.charAt(cantDigResultPot - 1));

				//Comparativa inputUserNum == Ultima digito de resultado potencia
				System.out.println((igualDigito = numInput == ultDig1_resultPot) ? ("Es automórfico") : ("No es automórfico"));
			}
			else //digitos == 2
			{
				//Cantidad digitos de Resultado Potencia
				cantDigResultPot = sResultPot.length();

				//Extraer ultima digito de Resultado Potencia "String" a variable int ultDig_resultPot
				ultDig1_resultPot = Character.getNumericValue(sResultPot.charAt(cantDigResultPot - 1));
				//System.out.println("\nultima digito de Resultado: "+ultDig1_resultPot);

				//Extraer pentiultima digito de Resultado Potencia "String" a variable int ultDig_resultPot
				ultDig2_resultPot = Character.getNumericValue(sResultPot.charAt(cantDigResultPot - 2));
				//System.out.println("pentiultima digito de Resultado: " + ultDig2_resultPot);

				// Int ultDig1_resultPot y ultDig2_resultPot a String variable ultimos2DigitosResPot
				ultimos2DigitosResPot = String.valueOf(ultDig2_resultPot + "" + ultDig1_resultPot);
				//System.out.println("ultDig1_resultPot y ultDig2_resultPot: "+ ultimos2DigitosResPot);

				//String a int ultimos2Dig
				ultimos2Dig = Integer.parseInt(ultimos2DigitosResPot);

				//Comparativa inputUserNum == Ultima digito de resultado potencia
				//igualDigito = numInput == ultimos2Dig;
				//System.out.println("   igualDigito: " + igualDigito);
				System.out.println((igualDigito = numInput == ultimos2Dig) ? ("Es automórfico") : ("No es automórfico"));
			}

			if (sResultPot.length() >= 19) {
				System.out.println("\nLo siento, no puede mostrar más " + i + " de exponente. Digitos esta fuera de limite");
				break;
			}
		}
	}

}

File: VSProjects/Actividad12/Actividad12.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Actividad12
{
	class Actividad12
	{
		static void Main(string[] args)
		{
			int numInput, digitos, ultDig1_resultPot, ultDig2_resultPot, ultimos2Dig, cantDigResultPot, repInicio;
			String numInputText, sResultPot, ultimos2DigitosResPot = "", msg = "";
			long resultPot;

				//Ingresa Usuario
				Console.WriteLine("Ingrese un numero:");
				numInput = Int32.Parse(Console.ReadLine());
				numInputText = numInput + "";
				digitos = numInputText.Length;

				//CalcularPotencia
				msg += "Base | Exponente | Resultado";
				for (int i = 1; i <= numInput; i++)
				{
					resultPot = (long)Math.Pow(numInput, i);
					sResultPot = "" + resultPot;
					msg += "\n" + numInput + "    | " + i + "\t    | " + resultPot + " | ";

					if (digitos == 1)
					{
						//Cantidad digitos de Resultado Potencia
						cantDigResultPot = sResultPot.Length;

						//Extraer ultima digito de Resultado Potencia "String" a variable ultDig_resultPot
						ultDig1_resultPot = (int)char.GetNumericValue((sResultPot[cantDigResultPot - 1]));

						//Comparativa inputUserNum == Ultima digito de resultado potencia
						msg += (numInput == ultDig1_resultPot) ? (" Es automórfico") : (" No es automórfico");

					}
					else //digitos == 2
					{
						//Cantidad digitos de Resultado Potencia
						cantDigResultPot = sResultPot.Length;

						//Extraer ultima digito de Resultado Potencia "String" a variable int ultDig_resultPot
						ultDig1_resultPot = (int)char.GetNumericValue(sResultPot[cantDigResultPot - 1]);

						//Extraer pentiultima digito de Resultado Potencia "String" a variable int ultDig_resultPot
						ultDig2_resultPot = (int)char.GetNumericValue(sResultPot[cantDigResultPot - 2]);

						// Int ultDig1_resultPot y ultDig2_resultPot a String variable ultimos2DigitosResPot
						ultimos2DigitosResPot = ultDig2_resultPot + "" + ultDig1_resultPot;

						//String a int ultimos2Dig
						ultimos2Dig = Int32.Parse(ultimos2DigitosResPot);

						//Comparativa inputUserNum == Ultima digito de resultado potencia
						msg += (numInput == ultimos2Dig) ? (" Es automórfico") : (" No es automórfico");
					}

					if (sResultPot.Length >= 19)
					{
						msg += "\n\nLo siento,\nno puede mostrar más " + i + " de exponente. Digitos esta fuera de limite";
						break;
					}

				}//Fin CalcularPotencia - verificiarAutomorfico
				Console.WriteLine(msg);

			Console.WriteLine("Presione una tecla para continuar...");
			Console.ReadKey(true);
		}

	}
}
